code/modules/mob/living/simple_animal/hostile/carp.dm

```diff
@@ -83,10 +85,23 @@
 		else
 			make_tameable()
 
+/mob/living/simple_animal/hostile/carp/revive(full_heal, admin_revive)
+	if (tamed)
+		var/datum/weakref/friendref = ai_controller.blackboard[BB_HOSTILE_FRIEND]
+		tamed(friendref.resolve())
+	return ..()
+
+/mob/living/simple_animal/hostile/carp/death(gibbed)
+	if (tamed)
+		can_buckle = FALSE
+		unbuckle_all_mobs()
+	return ..()
```

Comment:
Is there a reason why carps are snowflaked to unbuckle on death instead of handling it in the ridable element's `Detach()`?

Reply:
Err nope, ill do that